src/formatDate.js

```diff
@@ -50,7 +50,28 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = date.split(`${fromFormat[3]}`);
+  const dateItems = {};
+  let newDate = '';
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    dateItems[fromFormat[i]] = dateParts[i];
+  }
+
+  if (dateItems.hasOwnProperty('YY')) {
+    dateItems.YYYY = dateItems.YY < 30
+      ? '20' + dateItems.YY : '19' + dateItems.YY;
+  } else {
+    dateItems.YY = dateItems.YYYY.slice(2, 4);
+  }
+
+  for (let i = 0; i < toFormat.length; i++) {
+    if (dateItems.hasOwnProperty(toFormat[i])) {
+      newDate += dateItems[toFormat[i]] + `${toFormat[3]}`;
+    }
```

Comment:
some magic happens here. Let's create variables for `dateItems[toFormat[i]]` and `toFormat[3]`